doc/guides/smf/index.rst

```diff
@@ -0,0 +1,267 @@
+.. _smf:
+
+State Machine Framework
+#######################
+
+Overview
+========
+
+The State Machine Framework (SMF) is an application agnostic framework that
+provides an easy way for developers to integrate state machines into their
+application. The framework can be added to any project by including the
+CONFIG_SMF=y into the projects prj.conf file.
+
+State Creation
+==============
+
+A state is represented by three functions, where one function implements the
+Entry actions, another function implements the Run actions, and the last
+function implements the Exit actions. The prototype for these functions is as
+follows: **void funct(void *obj)**, where the **obj** parameter is a user
```

Comment:
Please read through the documentation guidelines and make adjustments: https://docs.zephyrproject.org/latest/contribute/documentation/index.html

E.g. this should be using double backticks for inline code literals as follows:

```
follows: ``void funct(void *obj)`` [...]
```

Reply:
Done